src/ck-core/cklocation.C

```diff
@@ -2509,13 +2510,15 @@ int CkLocMgr::deliverMsg(CkArrayMessage *msg, CkArrayID mgr, CmiUInt8 id, const
 #if CMK_LBDB_ON
   if ((idx || compressor) && type==CkDeliver_queue && !(opts & CK_MSG_LB_NOTRACE) && lbmgr->CollectingCommStats())
   {
+    // LB deals in IDs with collection information only when CMK_GLOBAL_LOCATION_UPDATE
+    // is enabled, so add the group information if so.
 #if CMK_GLOBAL_LOCATION_UPDATE
-    CmiUInt8 locMgrGid = thisgroup.idx;
-    id = ck::ObjID(id).getElementID();
-    id |= locMgrGid << ck::ObjID().ELEMENT_BITS;
+    const CmiUInt8 lbObjId = ck::ObjID(getGroupID(), id).getID();
```

Comment:
```suggestion
    const CmiUInt8 lbObjId = ck::ObjID(thisgroup, id).getID();
```

Reply:
I think it is slightly clearer if the variable is used directly so it's obvious that you're passing in the ID of this location manger group.